modules/cassandrabackend/cassandrabackend.cc

```diff
@@ -0,0 +1,324 @@
+			 soadata.db = this;
+			 //Data format :: domain_id nameserver	hostname    serial refresh retry expiry default_ttl
+			 std::string data = backendRecord.getRecord();
+			 boost::char_separator<char> delimiter("#");
+			 boost::tokenizer<boost::char_separator<char> > tokens(data, delimiter);
+			 int rec_index  = 0;
+			 BOOST_FOREACH(std::string const& token, tokens)
+			 {
+				 switch(rec_index) {
+				 	 case 0:
+				 		 soadata.domain_id = (uint32_t)atoi(token.c_str());
+				 		 break;
+				 	case 1:
+						 soadata.nameserver = token.c_str();
+						 break;
+				 	case 2:
+						 soadata.hostmaster = token.c_str();
+						 break;
+				 	case 3:
+						 soadata.serial = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 4:
+						 soadata.refresh = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 5:
+						 soadata.retry = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 6:
+						 soadata.expire = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 7:
+						 soadata.default_ttl = (uint32_t)atoi(token.c_str());
+						 break;
+				 	default:
+				 		L << Logger::Info << "[CassandraBackend]Extra params while parsing" << endl;
+				 }
+				 soadata.ttl = backendRecord.getTtl();
+				 rec_index++;
+			 }
+			 L << Logger::Info << "[CassandraBackend] SOA record true"<< endl;
+			 L << Logger::Info << "[CassandraBackend] SOA serial "<<soadata.serial<<" refresh "<<soadata.refresh<<" retry "<<soadata.retry
+					 <<" expire "<<soadata.expire<<" default_ttl "<<soadata.default_ttl<<" domain_id "<<soadata.domain_id<<" ttl "<<soadata.ttl
+					 <<" nameserver "<<soadata.nameserver<<" hostmaster "<<soadata.hostmaster<<endl;
+			 clear();
+			 return true;
+		 }
+	}
+	 L << Logger::Info << "[CassandraBackend] SOA record false"<< endl;
+	 clear();
+	 return false;
+
+  }
+
+  void populatedefaults(SOAData &soadata) {
+	 soadata.db = this;
+	 soadata.serial = 0;
+	 soadata.refresh = 10;
+	 soadata.retry = 10;
+	 soadata.expire = 10;
+	 soadata.default_ttl = 10;
+	 soadata.domain_id = 10;
+	 soadata.ttl = 10;
+	 soadata.nameserver = "ns1.default.pdns.com";
+	 soadata.hostmaster = "hm1.default.pdns.com";
+  }
+
+  void lookup(const QType &type, const string &qdomain, DNSPacket *p, int zoneId)
+  {
+    queryType = type.getName();
+    domain = qdomain;
+    recordIndex = 0;
+    totalSize = 0;
+    L << Logger::Info << "[CassandraBackend] Recieved query for "<<queryType<< " " <<domain<< " "<< endl;
+
+    fetchdata();
+
+  }
+
+  void fetchdata()
+  {
+	  L << Logger::Info << "[CassandraBackend] Calling cassandradbmanger" << endl;
+	  cassandradbmanager *sc1 = cassandradbmanager::getInstance();
+	  const char* query = "SELECT domain, recordmap, creation_time FROM pdns.domain_lookup_records WHERE domain = ?";
+	  L << Logger::Info << "[CassandraBackend] SELECT domain, recordmap, creation_time FROM pdns.domain_lookup_records WHERE domain = "<<domain<< endl;
```

Comment:
This seems bit excessive logging, use DLOG() to disable these unless debug compile is made?